sponsorship_compassion/model/contracts.py

```diff
@@ -63,6 +75,42 @@ def _get_contract_from_invoice(invoice_obj, cr, uid, invoice_ids,
 
         return list(res)
 
+    def _cancel_invoices(
+            self, cr, uid, partner_id, date_invoice, context=None):
+        invoice_line_obj = self.pool.get('account.invoice.line')
+        invoice_obj = self.pool.get('account.invoice')
+        invoice_ids = invoice_obj.search(
+            cr, uid,
+            [('partner_id', '=', partner_id),
+             ('state', '=', 'open'),
+             ('date_invoice', '<', date_invoice)
+             ],
+            context=context)
+
+        for invoice in invoice_obj.browse(cr, uid, invoice_ids, context):
+            invoice_line_ids = invoice_line_obj.search(
+                cr, uid,
+                [('invoice_id', '=', invoice.id)],
+                context=context)
+            invoice_lines = invoice_line_obj.browse(
+                cr, uid,
+                invoice_line_ids,
+                context)
+            contract_ids = [
+                invoice_line.contract_id.id for invoice_line in invoice_lines]

```

Comment:
add check if invoice_line.contract_id, otherwise this line will fail if there are some invoice lines unlinked to a contract_id.